src/library/scala/util/Random.scala

```diff
@@ -38,7 +39,10 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    *  array.
    */
   def nextBytes(bytes: Array[Byte]): Unit = { self.nextBytes(bytes) }
-
+  
+  /** Generates `n` random bytes and returns them in a new array. */
+  def nextBytes(n: Int): Array[Byte] = Array.ofDim[Byte](n).tap(self.nextBytes)
```

Comment:
I know it's tempting to use new features, but this isn't the place!

This is exactly the kind of thing that `tap` should only be used for if you've verified in the bytecode that it's fully erased, or benchmarked to make sure it doesn't have a significant performance impact on small byte arrays.

Please use known high-performance alternatives instead, e.g.

```
val bytes = new Array[Byte](n)
nextBytes(bytes)
bytes
```

(or use `match`, which is optimized by the compiler in all cases like this, potentially unlike `tap`).

Reply:
On the other hand, this generally shouldn't be a hot method -- you generally want to use re-use an array as a buffer in a loop (e.g. use the other method, or use a new method with a "buffer" signature `def nextBytes(buffer: Array[Byte], startOffset: Int, n: Int): Unit`, though there is no backing for this in the JDK) if this is hot to prevent the allocation

Reply:
@martijnhoekstra - I agree, it generally shouldn't.  But part of the job of writing a heavily-used library should be to help keep users from having to think about how hot something is.  One can only do so much, but if we can take on the burden of thinking about performance so others don't have to as often, it's usually a big win overall.